Documentation/General/emergentArchitectureDesign/emergentArchitectureDesign.tex

```diff
@@ -137,15 +141,26 @@ \section{Persistent data management}
 Moodcat has to handle both song data and chatroom data.
 
 \par
-For the song we store the basic properties, like artist, song title, genre and duration.
-Furthermore each song has its own \gls{valence} and \gls{arousal} value, which is computed and corrected by the music matching algorithm.
-The algorithm uses some of the audio features of the track for the computation of the valence and arousal values, these features are stored with the song.
+We try do deduplicate artists from the Soundcloud data as much as possible. We identify artists in our database by an unique identifier. Besides this identifier, we store the name for the artist.
+
+\par
+For the song we store the basic properties, like artist, song title and duration.
+A song has an unique id in our system and also an id from \gls{Soundcloud}, so we can fetch data from Soundcloud.
+We also include an url to display the artwork of a song.
+
+Furthermore each song has its own \gls{valence} and \gls{arousal} value, this is updated when users classify the song.
+This classification will be done in the Music Matching Algorithm as described above. So, we also have keep count of relative amount of up-votes compared to the down-votes.
```

Comment:
'In order to prevent incorrect or unwanted classifications, we also have to store the netto value of up-votes compared to the down-votes.'